### PR TITLE
Update docs to reflect charts->phalanx migration

### PR DESCRIPTION
**Pull request by athornton**

### CODE REVIEW

File: docs/arch/repository.rst [hunk 1]
```diff
@@ -47,21 +44,21 @@ Chart versioning
...
-Charts maintained in lsst-sqre/charts follow this convention (for the most part).
+Charts maintained as first-party charts in Phalanx follow this convention (for the most part).
 Most upstream charts also follow this convention, but some require explicitly changing version numbers in ``values-*.yaml``.
 
 In general, we pin the version of the chart to deploy in the ``dependencies`` metadata of the top-level chart.
 This ensures deterministic cluster configuration and avoids inadvertently upgrading services.
 However, for services still under development, we sometimes use a floating dependency to reduce the number of pull requests required when iterating, and then switch to a pinned version once the service is stable.
 
-There is currently no mechanism to deploy different versions of a chart in different environments.
+There is currently no general mechanism to deploy different versions of a chart in different environments, as appVersion is set in ``Chart.yaml``.
```
Comment (jonathansick) [inline docs/arch/repository.rst:62]: `appVersion` can be overridden with something like a `image.tag` value in a values file, but that doesn't really get to the point that all environments use the same chart fundamentally.

We do have the ability to test new chart versions by deploying from branches through. Can we link to docs for that? Do those docs exist?

**Review by jonathansick — COMMENTED** [review]
Can you take another look at https://phalanx.lsst.io/v/DM-35703/service-guide/create-service.html#create-the-helm-chart ? There's still a number of outdated things in there. Generally, I wonder if a lot of this section can either be dropped or moved to `add-service.rst` (so the page focuses on creating the app itself and then `add-service.rst` focuses on the Helm chart aspect)

File: docs/arch/repository.rst (hunk 1 above)
Comment (athornton) [reply docs/arch/repository.rst:62]: Not sure if those docs exist, but if not, I'll write some.  I'll also restructure create-service.

Comment (athornton) [reply docs/arch/repository.rst]: I've taken a crack at it.

**Comment by athornton** [comment]
I incorporated @jonathansick 's suggestions and I think it's ready for another look.

File: docs/arch/repository.rst [hunk 2]
```diff
@@ -59,6 +59,8 @@ In general, we pin the version of the chart to deploy in the ``dependencies`` me
...
-We will probably need a mechanism to do this eventually, and have considered possible implementation strategies, but have not yet started on this work.
-In the meantime, we disable automatic deployment in Argo CD so there is a human check on whether a given chart is safe to deploy in a given environment.
+There is currently no generic mechanism to deploy different versions of a chart in different environments, as appVersion is set in ``Chart.yaml``.
+
+That does not mean that rolling out a new version is all-or-nothing: you have a couple of different options for testing new versions.  The easiest is to modify the appVersion in ``Chart.yaml`` on your development branch and then use ArgoCD to deploy the application from the branch, rather than ``master``, ``main``, or ``HEAD`` (as the case may be).  This will cause the application resource in the ``science-platform`` app to show as out of sync, which is indeed correct, and a helpful reminder that you may be running from a branch when you forget and subsequently rediscover that fact weeks later.
```
Comment (jonathansick) [inline docs/arch/repository.rst:64]: This is good to have. I think at some point we'd like to refactor this into a separate a how-to page specifically about branch-based so it's easier to find in the moment. I wouldn't worry about tackling this before merging though.

File: docs/service-guide/create-service.rst [hunk 3]
```diff
@@ -32,72 +32,16 @@ If you are using the FastAPI template, tagging in this fashion is required since
...
+
+If you use ghcr.io as your repository (which is the FastAPI template default) you require a repository secret named ``GITHUB_TOKEN``.
+If you are using Docker Hub you must create two secrets in your new GitHub repository, ``DOCKER_USERNAME`` and ``DOCKER_TOKEN``.
+``DOCKER_USERNAME`` should be set to the Docker Hub username of the account that will be pushing the new Docker images.
 ``DOCKER_TOKEN`` should be set to a secret authentication token for that account.
 We recommend creating a separate token for each GitHub repository for which you want to enable automatic image publication, even if they all use the same username.
 
-You may need to have a Docker Pro or similar paid DockerHub account.
+If using Docker Hub You may need to have a Docker Pro or similar paid Docker Hub account.
 Alternately, you can contact SQuaRE to set up Docker image publication using our Docker account.
```
Comment (jonathansick) [inline docs/service-guide/create-service.rst:47]: Maybe close out this page with a one-sentence paragraph that then links to the `add-service.rst` page. Something along the lines of:

> Next, deploy your service by create a Helm chart in Phalanx.

File: docs/service-guide/create-service.rst [hunk 4]
```diff
@@ -32,72 +32,16 @@ If you are using the FastAPI template, tagging in this fashion is required since
...
 If your image must be stored in a private container registery, the credentials for that registry must be added to the pull secret.
 
 If you use the FastAPI service template, a ``Dockerfile`` will be created as part of the new repository template, and GitHub Actions will be set up in the new repository to build and push new Docker images for tagged releases.
-To enable this workflow, you must create two secrets in your new GitHub repository, ``DOCKER_USERNAME`` and ``DOCKER_TOKEN``.
-``DOCKER_USERNAME`` should be set to the DockerHub username of the account that will be pushing the new Docker images.
+
+If you use ghcr.io as your repository (which is the FastAPI template default) you require a repository secret named ``GITHUB_TOKEN``.
```
Comment (jonathansick) [inline docs/service-guide/create-service.rst:40]: Note that `GITHUB_TOKEN` is built into GitHub Actions (i.e. you don't configure it, it just exists for every workflow). 

So something along the lines of:

> If you use ghcr.io as your repository (which is the FastAPI template default) you can use GitHub's built-in ``GITHUB_TOKEN``, you don't need to add an additional secret.

**Review by jonathansick — APPROVED** [review]

File: docs/service-guide/create-service.rst (hunk 4 above)
Comment (athornton) [reply docs/service-guide/create-service.rst]: Done.

File: docs/service-guide/create-service.rst (hunk 3 above)
Comment (athornton) [reply docs/service-guide/create-service.rst:49]: Done (also a pointer to the intermediate "add a secret" step).

File: docs/arch/repository.rst (hunk 2 above)
Comment (athornton) [reply docs/arch/repository.rst:64]: Thanks.  I agree that this is a future enhancement.

**Pull request merged**